aws/resource_aws_guardduty_filter.go

```diff
@@ -0,0 +1,489 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsGuardDutyFilter() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGuardDutyFilterCreate,
+		Read:   resourceAwsGuardDutyFilterRead,
+		Update: resourceAwsGuardDutyFilterUpdate,
+		Delete: resourceAwsGuardDutyFilterDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"detector_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(3, 64),
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 512),
+			},
+			"tags": tagsSchema(),
+			"finding_criteria": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"criterion": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"field": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringInSlice(criteriaFields(), false),
+									},
+									"condition": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"equals",
+											"not_equals",
+											"greater_than",
+											"greater_than_or_equal",
+											"less_than",
+											"less_than_or_equal",
+										}, false),
+									},
+									"values": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"action": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"NOOP",
+					"ARCHIVE",
```

Comment:
These should be replaced by the constants defined in the AWS SDK
```suggestion
					guardduty.FilterActionNoop,
					guardduty..FilterActionArchive,
```

Reply:
https://github.com/terraform-providers/terraform-provider-aws/commit/5d4dd0807276dabbfa23c926293cd4bfcf35da56